nf_core/lint/readme.py

```diff
@@ -40,7 +40,7 @@ def readme(self):
     # Check that there is a readme badge showing the minimum required version of Nextflow
     # [![Nextflow](https://img.shields.io/badge/nextflow%20DSL2-%E2%89%A521.10.3-23aa62.svg)](https://www.nextflow.io/)
     # and that it has the correct version
-    nf_badge_re = r"\[!\[Nextflow\]\(https://img\.shields\.io/badge/nextflow%20DSL2-%E2%89%A5([\d\.]+)-23aa62\.svg\)\]\(https://www\.nextflow\.io/\)"
+    nf_badge_re = r"\[!\[Nextflow\]\(https://img\.shields\.io/badge/nextflow%20DSL2-[\d\w%]+(\d{2}\.\d{1,2}\.\d{1,2})-23aa62\.svg\)\]\(https://www\.nextflow\.io/\)"
```

Comment:
This regex won't match the `!` versions. See https://regex101.com/r/P0827m/1.

The suggestion will match both. See https://regex101.com/r/P0827m/2
```suggestion
    nf_badge_re = r"\[!\[Nextflow\]\(https://img\.shields\.io/badge/nextflow%20DSL2-!?[\d\w%]+(\d{2}\.\d{1,2}\.\d{1,2})-23aa62\.svg\)\]\(https://www\.nextflow\.io/\)"
```

Reply:
But generally, the previous regex was better, because this one will fail to extract versions bigger than 99.

Reply:
I think it would be better to allow the different versions in options in a non-capturing grpup like so
```suggestion
    nf_badge_re = r"\[!\[Nextflow\]\(https://img\.shields\.io/badge/nextflow%20DSL2-!?(?:%E2%89%A5|%3E%3D)([\d\.]+)-23aa62\.svg\)\]\(https://www\.nextflow\.io/\)"
```

See https://regex101.com/r/P0827m/4